numpy/random/src/distributions/distributions.c

```diff
@@ -1450,22 +1469,34 @@ void random_bounded_uint64_fill(bitgen_t *bitgen_state, uint64_t off,
       out[i] = off;
     }
   } else if (rng <= 0xFFFFFFFFUL) {
-    uint32_t buf = 0;
-    int bcnt = 0;
-
     /* Call 32-bit generator if range in 32-bit. */
-    if (use_masked) {
-      /* Smallest bit mask >= max */
-      uint64_t mask = gen_mask(rng);
 
+    /*
+     * The 32-bit Lemire method does not handle rng=0xFFFFFFFF, so we'll
+     * call next_uint32 directly.  This also works when use_masked is True,
+     * so we handle both cases here.
+     */
+    if (rng == 0xFFFFFFFFUL) {
       for (i = 0; i < cnt; i++) {
-        out[i] = off + buffered_bounded_masked_uint32(bitgen_state, rng, mask,
-                                                      &bcnt, &buf);
+        out[i] = off + (uint64_t) next_uint32(bitgen_state);
```

Comment:
Is this code used in `RandomState`, and does this preserve the stream guarantee?  RS uses only masked, so intercepting before this might have been called could can an issue. 

Would it be safest to added a test in RandomState too with the same end points?

Reply:
This is tested by the test that was added in gh-14501: https://github.com/numpy/numpy/blob/20f10762fc20a29923b6943235c2577b8e3ac9f3/numpy/random/tests/test_randomstate_regression.py#L178